test/e2e/serviceloadbalancer_helpers.go

```diff
@@ -128,7 +128,7 @@ func waitForMetalLBServiceLoadBalancerToBeReadyInWorkloadCluster(
 
 	resources := make([]client.Object, len(cos))
 	for i := range cos {
-		resources[i] = cos[i]
+		resources[i] = cos[i].DeepCopy()
```

Comment:
Note: Consequence of passing around a slice of unstructured.Unstructured, instead of a slice of pointers to unstructured.Unstructured.